assets/js/main.js

```diff
@@ -72,3 +75,253 @@ $(document).ready(function() {
+        $('#header-auth-loading')
+            .removeClass('transition-in')
+            .addClass('transition-out')
+    }
+  }
+
+  // Header style toggle
+  function setHeaderStyle() {
+    const navTrigger = $('#nav-trigger')
+    if (navTrigger.length) {
+        const cutoff = 40
+        const distance = $(window).scrollTop() - navTrigger.offset().top - cutoff
+        if (distance > 10 && !$('#header').hasClass('white')) {
+            $('#header').addClass('white')
+        }
+        if (distance < 10 && $('#header').hasClass('white')) {
+            $('#header').attr('class', window.headerIntialClasses)
+        }
+    }
+  }
+
+  // Email validation
+  function validateEmail(email) {
+    var re = new RegExp('[a-zA-Z0-9_\\.\\+-]+@[a-zA-Z0-9-]+\\.[a-zA-Z0-9-\\.]+')
+    return re.test(String(email).toLowerCase())
+  }
+
+  // Cookies
+  function setCookie(name, value, days) {
+    var expires = ''
+    if (days) {
+        var date = new Date()
+        date.setTime(date.getTime() + days * 24 * 60 * 60 * 1000)
+        expires = '; expires=' + date.toUTCString()
+    }
+    document.cookie = `${name}=${value || ''}${expires}; path=/; domain=${window.location.hostname}; SameSite=Lax`
+  }
+  function getCookie(name) {
+    var nameEQ = name + '='
+    var ca = document.cookie.split(';')
+    for (var i = 0; i < ca.length; i++) {
+        var c = ca[i]
+        while (c.charAt(0) === ' ') c = c.substring(1, c.length)
+        if (c.indexOf(nameEQ) === 0) return c.substring(nameEQ.length, c.length)
+    }
+  }
+  function setCookieNotification() {
+    var accepted = getCookie('codemagic_accept_cookies')
+    if (!accepted) {
+        $('#cookie-notice').show()
+    }
+  }
+
+  function activeUserChanges() {
+    $(document).ready(function() {
+        $('[ad-type*="product"]').hide()
+    })
+    $('#footer-cta .btn')
+        .text('Return to app')
+        .attr('href', '{{ site.Param "appURL" }}/apps')
+  }
+  function visitorChanges() {
+    $(document).ready(function() {
+        $('[ad-type*="product"]').show()
+    })
+    $('#footer-cta .btn')
+        .text('Get Started Now')
+        .attr('href', '{{ site.Param "appURL" }}/signup')
+  }
+})
+
+let lastScrollPosition = 0
+const sidebar = $('#sidebar')
+const header = $('#header')
+const docsMenu = $('#docs-menu')
+const contentWrap = $('#content-wrap')
+
+if ($(window).scrollTop() === 0) {
+  header.css('top', 0)
+  sidebar.css('top', header.innerHeight())
+}
+
+$(window).on('scroll', function() {
+    const currentScrollPosition = $(window).scrollTop()
+    window.scrollingDown = currentScrollPosition > lastScrollPosition
+    lastScrollPosition = currentScrollPosition
+})
+
+$(window).on('load scroll resize', function() {
+  if ($(window).width() < desktopScreenWidth) {
+    if (window.scrollingDown) {
+      sidebar.css('top', 0)
+      header.css('top', -header.innerHeight())
+      docsMenu.css('top', sidebar.innerHeight())
```

Comment:
This surely isn't right. Sidebar height is basically the viewport height and when you apply `top` that large, docs menu will not be visible. You can see this bug simply by resizing your viewport to below 1001px.

Reply:
Not sure if I understood correctly, probably do you mean when changing from below 1001px to higher?
I changed `$(window).width()` was giving value on 15px lower actual window size, so CSS changes were applied, js - no.
Fixed replacing it to `window.innerWidth`